frontend/src/stack/create/create-by-form.html

```diff
@@ -10,7 +10,7 @@
             <div class="form-control">
                 <input id="createByForm-Namespace" type="text" data-ng-model="stackCreateByFormCtrl.form.Namespace"
                        data-required="required" name="Name" focus-me
-                       data-ng-pattern="/^[A-Za-z0-9]+$/">
+                       data-ng-pattern="/^\w[\w_.-]+\w$/">
```

Comment:
Here the **namespace** is **Stack Name** instead, not **service name** .

At the same time, the service name format is StackNamespace_ServiceName deployed by crane. We have to avoid the charactor *_* in stack namespace pattern , in case of the split confuse.

So make data-ng-pattern="/^\w[\w.-]?\w$/" , IMO.